packages/api/src/utils/billing.js

```diff
@@ -1,24 +1,40 @@
 /* eslint-disable no-void */
 
+import { InvalidTosAgreementError, TosAgreementRequiredError } from '../../src/errors.js'
+
 /**
  * @typedef {import('./billing-types').StoragePriceName} StoragePriceName
  */
 
+/**
+ * @param {import('./billing-types').Agreement} agreement
+ * @returns {boolean}
+ */
+function isAgreement (agreement) {
+  return agreement === 'web3.storage-tos-v1'
+}
+
 /**
  * Save a user's payment settings
  * @param {object} ctx
  * @param {import('./billing-types').BillingService} ctx.billing
  * @param {import('./billing-types').CustomersService} ctx.customers
  * @param {import('./billing-types').SubscriptionsService} ctx.subscriptions
+ * @param {import('./billing-types').TermsOfServiceService} ctx.termsOfService
  * @param {import('./billing-types').BillingUser} ctx.user
  * @param {object} paymentSettings
  * @param {Pick<import('./billing-types').PaymentMethod, 'id'>} paymentSettings.paymentMethod
  * @param {import('./billing-types').W3PlatformSubscription} paymentSettings.subscription
+ * @param {import('./billing-types').Agreement} paymentSettings.agreement
  * @param {import('./billing-types').UserCreationOptions} [userCreationOptions]
  */
 export async function savePaymentSettings (ctx, paymentSettings, userCreationOptions) {
-  const { billing, customers, user } = ctx
+  if (!paymentSettings.agreement) throw new TosAgreementRequiredError()
```

Comment:
I like how you added these nice semantic errors here.
On the other hand, If this is always required, can we just make it required in the `PaymentSettings` type so it's not even compilable to call this function with an invalid value and get a runtime error?

(the caller of `savePaymentSettings` would then be responsible for getting an appropriate value, and could still throw the exact errors you did, but it might have more specific context to use to throw the error e.g. with a really useful error message).